drivers/axi_core/axi_adc_core/axi_adc_core.c

```diff
@@ -150,6 +150,118 @@ void axi_adc_idelay_set(struct axi_adc *adc,
 
 }
 
+/***************************************************************************//**
+ * @brief axi_adc_delay_set
+*******************************************************************************/
+int32_t axi_adc_delay_set(struct axi_adc *adc,
+			  uint32_t no_of_lanes,
+			  uint32_t delay)
+{
+	uint32_t i;
+	uint32_t rdata;
+	uint32_t pcore_version;
+
+	axi_adc_read(adc, 0x0, &pcore_version);
+	pcore_version >>= 16;
+	if (pcore_version < 9) {
+		printf(" pcore_version is : %d\n\r", pcore_version);
+		printf(" DRIVER DOES NOT SUPPORT PCORE VERSIONS OLDER THAN 10 !");
+		return FAILURE;
+	} else {
+		for (i = 0; i < no_of_lanes; i++) {
+			axi_adc_write(adc,((0x200 + i)*4), delay);
```

Comment:
I think you can use `AXI_ADC_REG_DELAY` define here:
```suggestion
			axi_adc_write(adc,AXI_ADC_REG_DELAY(i), delay);
```
Same below.

Reply:
The `AXI_ADC_REG_DELAY` has different definition, therefore I think it won't work properly if i use it.

The macro is used for `axi_adc_idelay_set` function which is outside of the commit's scope.

Reply:
My bad, you are right here, `AXI_ADC_REG_DELAY` can be used. Will do the change.